r2r/parsers/media/pdf_parser.py

```diff
@@ -24,14 +24,26 @@ async def ingest(self, data: DataType) -> AsyncGenerator[str, None]:
         """Ingest PDF data and yield text from each page."""
         if isinstance(data, str):
             raise ValueError("PDF data must be in bytes format.")
-
+        import unicodedata
         pdf = self.PdfReader(BytesIO(data))
         for page in pdf.pages:
             page_text = page.extract_text()
             if page_text is not None:
                 page_text = "".join(
-                    filter(lambda x: x in string.printable, page_text)
-                )
+                    filter(lambda x: (
+                      unicodedata.category(x) in ['Ll', 'Lu', 'Lt', 'Lm', 'Lo', 'Nl', 'No'] or  # Keep letters and numbers
+                        '\u4E00' <= x <= '\u9FFF' or  # Chinese characters
+                        '\u0600' <= x <= '\u06FF' or  # Arabic characters
+                        '\u0400' <= x <= '\u04FF' or  # Cyrillic letters
+                        '\u0370' <= x <= '\u03FF' or  # Greek letters
+                        '\u0E00' <= x <= '\u0E7F' or  # Thai
+                        '\u3040' <= x <= '\u309F' or  # Japanese Hiragana
+                        '\u30A0' <= x <= '\u30FF' or  # Katakana
+                        # Other printable characters
+                        x in string.printable
+                    ), page_text)
```

Comment:
Can you add a flag called multilingual to enable this behavior and pass it to r2r.json config?